server/verify.go

```diff
@@ -245,7 +245,7 @@ func validateAKPub(ak crypto.PublicKey, opts VerifyOpts) error {
 
 func validateAKCert(akCert *x509.Certificate, opts VerifyOpts) (*pb.MachineState, error) {
 	if len(opts.TrustedRootCerts) == 0 {
-		return nil, validateAKPub(akCert.PublicKey.(crypto.PublicKey), opts)
+		return &pb.MachineState{}, validateAKPub(akCert.PublicKey.(crypto.PublicKey), opts)
```

Comment:
I think `validateAKPub` should return `(*pb.MachineState, error)` instead of just `error`. That way we always get a non-nil MachineState xor a non-nil error. 

If for some reason we don't do this, we should explain why this is needed, since someone might try to "simplify" this.

Reply:
added